students/AurelPerianu/Lesson6/test_mailroom_4.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+import unittest
+import mailroom_4 as mailroom
+import os
+
+
+class MailroomTest(unittest.TestCase):
+
+    def test_donor_amount(self):
+        self.assertEqual(mailroom.donors['Donor A'], [3580, 34124.31, 7654])
+
+    def test_list_donors(self):
+        del mailroom.donors['x']
+        self.assertEqual(list(mailroom.donor_list()), ['Donor A', 'Donor B',
+                         'Donor C', 'Donor D', 'Donor E', 'Donor F'])
+
+    def test_report(self):
+        self.assertTrue(mailroom.report())
```

Comment:
I would recommend refactoring the `report` function to return the report as a string. That way you could test the dynamically generated output of your program